Invoro.Api/Invoro.Api/src/Services/FeaturesService.cs

```diff
@@ -32,11 +33,11 @@ await this.FeaturesCollection
 
         public async Task<IEnumerable<string>> GetVotedFeaturesByUser(string userId)
         {
-            IEnumerable<string> result = this.FeaturesVotesCollection.AsQueryable<FeatureVote>()
-                .Where(featureVote => featureVote.UserId == userId)
-                .Select(feature => feature.FeatureId);
+            IMongoQueryable<string> result = this.FeaturesVotesCollection.AsQueryable<FeatureVote>().
+                Where(featureVote => featureVote.UserId == userId).
+                Select(feature => feature.FeatureId);
 
-            return result;
+            return await result.ToListAsync();
```

Comment:
ToListAsync need to get CancellationToken, you have it on The BaseService.
It will cancel the mongo requeust in case the request to server has cancelled.